lang/cs/Org.Apache.REEF.Tests/Functional/Bridge/TestSecurityToken.cs

```diff
@@ -0,0 +1,238 @@
+                .Build();
+
+            var result = reefClient.SubmitAndGetJobStatus(jobSubmission);
+            var state = PullFinalJobStatus(result);
+            Logger.Log(Level.Info, "Application final state : {0}.", state);
+            Assert.Equal(FinalState.SUCCEEDED, state);
+        }
+
+        /// <summary>
+        /// Use HelloDriver in the test
+        /// </summary>
+        /// <returns>Return driver configuration.</returns>
+        private IConfiguration GetDriverConfig()
+        {
+            return DriverConfiguration.ConfigurationModule
+                .Set(DriverConfiguration.OnEvaluatorAllocated, GenericType<HelloDriver>.Class)
+                .Set(DriverConfiguration.OnDriverStarted, GenericType<HelloDriver>.Class)
+                .Build();
+        }
+
+        /// <summary>
+        /// Pull job final status until the Job is done
+        /// </summary>
+        /// <param name="jobSubmitionResult"></param>
+        /// <returns>Return final state of the job.</returns>
+        private FinalState PullFinalJobStatus(IJobSubmissionResult jobSubmitionResult)
+        {
+            int n = 0;
+            var state = jobSubmitionResult.FinalState;
+            while (state.Equals(FinalState.UNDEFINED) && n++ < 100)
+            {
+                Thread.Sleep(3000);
+                state = jobSubmitionResult.FinalState;
+            }
+            return state;
+        }
+
+        /// <summary>
+        /// Get runtime configuration.
+        /// Bind tokens to YarnClientCnfiguration.
+        /// </summary>
+        /// <returns>Return runtime configuration for multiple tokens.</returns>
+        private static IConfiguration GetRuntimeConfigurationForMultipleTokens()
+        {
+            var yarnClientConfigModule = YARNClientConfiguration.ConfigurationModule;
+            foreach (var t in CreateTestTokens())
+            {
+                yarnClientConfigModule = yarnClientConfigModule.Set(YARNClientConfiguration.SecurityTokenStr, t);
+            }
+
+            return Configurations.Merge(yarnClientConfigModule.Build(), TcpPortConfig());
+        }
+
+        /// <summary>
+        /// Get runtime configuration.
+        /// Bind token to YarnClientCnfiguration.
+        /// </summary>
+        /// <returns>Return runtime configuration for single token.</returns>
+        private static IConfiguration GetRuntimeConfigurationForSingleToken()
+        {
+            var yarnClientConfigModule = YARNClientConfiguration.ConfigurationModule
+                .Set(YARNClientConfiguration.SecurityTokenStr, CreateTestToken());
+
+            return Configurations.Merge(yarnClientConfigModule.Build(), TcpPortConfig());
+        }
+
+        /// <summary>
+        /// Get runtime configuration and token with old approach
+        /// </summary>
+        /// <returns>Return runtime configuration for old approach.</returns>
+        /// TODO: [JIRA REEF-1887] Will remove after the old approach is removed. 
```

Comment:
use `[System.Obsolete("...")]` method attribute; mention the version in which it should be removed

Reply:
Done